### PR TITLE
Connect api

### PR DESCRIPTION
**Pull request by alef-garrido**
### Hi there 👋🏻

Thanks for taking the time to review the project! I know that being a reviewer is not easy, so I appreciate your time and contributions.

### Changes implemented:

- Refactor `store` to use slice utility from redux-toolkit
- Combined reducers
- Added `devtools` enhancer to store
- Added thunk to middleware configuration object
- Added `api` middleware to handle API calls
- Added dispatching events to `api` middleware to enable 
- Refactor code to replace `axios` library with fetch API
- Fixed re-rendering bug refactoring `bookAdded` and `bookRemoved` reducers

### Final notes
This PR has blood and tears, please spare some directions/hints if there's any other change to make.

Is expected that this project meets the requirements. Please let me know if anything is left to be done or can be improved. Again, thanks for the review.

### CODE REVIEW

File: src/redux/books.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export const removeBook = (id) => apiCallBegan({
+  url: `${url}/${id}`,
+  method: 'DELETE',
+  body: JSON.stringify({ item_id: id }),
+  onStart: booksRequested.type,
+  onSuccess: booksReceived.type,
+  onError: booksRequestFailed.type,
+});
+
+export default slice.reducer;
```
Comment (medaminedev66) [inline src/redux/books.js]: - [x] It's the first time I see this approach. I love it. Only one issue I want to address. In Redux, we use actions to handle the state of the books array. The state books should be updated each time I add a book or remove a book. It means the book should be rendered in the layout instantly after adding it and it should be removed from the layouts after clicking on the remove button. Let's work on that.

File: src/redux/books.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable no-unused-vars */
```
Comment (medaminedev66) [inline src/redux/books.js]: [Optional] Rather than disable the linter strive to fix them as this makes you more accustomed to the style guide of the tech.

**Review by medaminedev66 — CHANGES_REQUESTED** [review]
Hi @alef-garrido 😃, :wave:, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- There are no linter errors ✔️

## Required Changes ♻️
- [ ] Pull Request should have a descriptive description telling others what changes are inside. Please update that.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Amrhub — APPROVED** [review]
# Approved 💯 ✅ 🥳 🎈 🎉 

Hi @alef-garrido, 👋🏻 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights ✅ 
Highly organized and clean code ✔️ 👨🏻‍💻 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**